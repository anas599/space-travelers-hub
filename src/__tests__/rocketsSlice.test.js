import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import RocketsFunction from '../pages/Rockets';

const mockStore = configureStore([]);
const description = 'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009....';

describe('RocketsFunction', () => {
  it('single rocket info to display', () => {
    const rocketsArr = [
      {
        id: '1',
        rocket_name: 'Falcon 1',
        description,
        reserved: false,
        flickr_images: [
          'https://live.staticflickr.com/764/22460724618_69d817d02a_b.jpg',
        ],
      },
    ];
    const initialState = {
      rockets: {
        rocketsArr,
      },
    };
    const store = mockStore(initialState);
    render(
      <Provider store={store}>
        <RocketsFunction />
      </Provider>,
    );
    const rocketName = screen.getByText('Falcon 1');
    const rocketDesc = screen.getByText(description);
    expect(rocketName).toBeInTheDocument();
    expect(rocketDesc).toBeInTheDocument();
    expect(store.getState()).toMatchSnapshot();
  });
});
