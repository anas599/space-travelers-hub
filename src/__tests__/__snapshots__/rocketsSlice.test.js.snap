// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketsFunction single rocket info to display 1`] = `
Object {
  "rockets": Object {
    "rocketsArr": Array [
      Object {
        "description": "The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009....",
        "flickr_images": Array [
          "https://live.staticflickr.com/764/22460724618_69d817d02a_b.jpg",
        ],
        "id": "1",
        "reserved": false,
        "rocket_name": "Falcon 1",
      },
    ],
  },
}
`;
